### PR TITLE
Add NumberInput component

### PR DESCRIPTION
**Pull request by jiri-kantor**
Component only allows a valid number to be entered. Number of decimals can be specified. It is based on TextInput and when an invalid value is entered, it emits an `onChange` event with a `null` value.

### CODE REVIEW

**Review by samuelg0rd0n — CHANGES_REQUESTED** [review]
I don't know if I'm doing something wrong but when I'm looking at it in Storybook, it doesn't matter what I write in the input, it keeps saying "Wrong number format". Can you take a look at it?

Screenshot 2022-02-14 at 10 57 03

**Comment by jiri-kantor** [comment]
I must've tested it with an outdated cache or something. I would swear it had worked, but when I tried it now, it didn't, because of unescaped backslashes.. I've changed the handling.

**Review by samuelg0rd0n — CHANGES_REQUESTED** [review]
There are several problems that I came across:

- negative numbers are not supported. I understand that in Purple Trading, it will be used primarily to input currency amounts so it doesn't make sense but phoenix components should be generic library and the component is called NumberInput and not PositiveNumberInput 😃 
- only decimal point is supported. Decimal points vs decimal commas are region specific. If we don't want to overcomplicate things, I'd like to at least support both.
- on mobile devices, there's currently an alphabet keyboard but we want a numeric one. Please use `inputmode="decimal"` when `maxDecimalCount > 0` and `inputmode="numeric"` otherwise.

File: src/components/NumberInput/utils.ts [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+// Limits to two decimals and strips any leading zeroes
+// We assume it is a valid numeric value (the isValidNumeric function returns true for it)
+export function getSanitizedValue(
+	value: string,
+	maxDecimalCount: number
+): {
+	stringValue: string
+	numberValue: number
+} {
+	console.log(value)
```
Comment (samuelg0rd0n) [inline src/components/NumberInput/utils.ts:14]: Forgotten console.log

**Review by samuelg0rd0n — APPROVED** [review]

**Pull request merged**